src/main/java/gift/domain/product/controller/ProductController.java

```diff
@@ -0,0 +1,101 @@
+
+import gift.domain.product.dao.ProductDao;
+import gift.domain.product.dto.ProductDto;
+import gift.domain.product.entity.Product;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+@RequestMapping("/products")
+public class ProductController {
+
+    private final ProductDao productDao;
+    private final Map<Long, Product> productRepository = new ConcurrentHashMap<>();
+    private final AtomicLong key = new AtomicLong(1);
+
+    public ProductController(ProductDao productDao) {
+        this.productDao = productDao;
+    }
+
+    @GetMapping("/new")
+    public String renderingNewForm() {
+        return "new-product";
+    }
+
+    @GetMapping("/edit/{productId}")
+    public String renderingEditForm(@PathVariable long productId, Model model) {
+        Optional<Product> product = productDao.findById(productId);
+
+        if (product.isEmpty()) {
+            return "error";
+        }
+        model.addAttribute("product", product.get());
+
+        return "edit-product";
+    }
+
+    @PostMapping
+    public String create(@ModelAttribute ProductDto productDto) {
+        Product product = productDto.toProduct();
+
+        Product savedProduct = productDao.insert(product);
+
+        return "redirect:/products/" + savedProduct.getId();
+    }
+
+    @GetMapping
+    public String readAll(Model model) {
+        List<Product> productList = productDao.findAll();
+        model.addAttribute("products", productList);
+        return "products";
+    }
+
+    @GetMapping("/{productId}")
+    public String readById(@PathVariable long productId, Model model) {
+        Optional<Product> product = productDao.findById(productId);
+
+        if (product.isEmpty()) {
+            return "error";
+        }
+        model.addAttribute("product", product.get());
+
+        return "product";
+    }
+
+    @PutMapping("/{productId}")
+    public String update(@PathVariable long productId, @ModelAttribute ProductDto productDto) {
+        Product product = productDto.toProduct();
+        product.setId(productId);
+
+        Optional<Product> updatedProduct = productDao.update(product);
+
+        return updatedProduct.map(value -> "redirect:/products/" + value.getId()).orElse("error");
```

Comment:
```suggestion
        return updatedProduct
        .map(value -> "redirect:/products/" + value.getId())
        .orElse("error");
```

혹시 Map을 하신 이유가 있나요?

위에처럼 if문으로 해도 되는데 통일시키면 어떨까요?

Reply:
별 다른 이유는 없습니다. IDE에 노란줄이 보기 싫어서 추천해주는 방향으로 수정했습니다..😅
말씀해주신 대로 통일시키는 게 좋을 것 같습니다!!